src/libraries/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Windows/X509Pal.PublicKey.cs

```diff
@@ -53,9 +74,13 @@ public AsymmetricAlgorithm DecodePublicKey(Oid oid, byte[] encodedKeyValue, byte
             }
         }
 
-        private static ECDsa DecodeECDsaPublicKey(CertificatePal certificatePal)
+        private static TAlgorithm DecodeECPublicKey<TAlgorithm>(
+            CertificatePal certificatePal,
+            Func<CngKey, TAlgorithm> factory,
+            Action<TAlgorithm, ECParameters> import)
+                where TAlgorithm : AsymmetricAlgorithm, new()
         {
-            ECDsa ecdsa;
+            TAlgorithm ecdsa;
```

Comment:
```suggestion
            TAlgorithm key;
```

(or something other than the name of one of the input algorithm types :smile:)